pkg/operator2/routercerts/controller.go

```diff
@@ -0,0 +1,252 @@
+package routercerts
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	v12 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	controllerWorkQueueKey           = "key"
+	conditionRouterCertsDegradedType = "RouterCertsDegraded"
+)
+
+// RouterCertsDomainValidationController validates that router certs match the ingress domain
+type RouterCertsDomainValidationController struct {
+	operatorClient v1helpers.OperatorClient
+	cachesToSync   []cache.InformerSynced
+	queue          workqueue.RateLimitingInterface
+	eventRecorder  events.Recorder
+
+	ingressLister v12.IngressLister
+	secretLister  v1.SecretNamespaceLister
+	secretName    string
+	routerName    string
+}
+
+func NewRouterCertsDomainValidationController(
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+	ingressLister v12.IngressLister,
+	secretLister v1.SecretNamespaceLister,
```

Comment:
take a route lister, so you can get the actual route being used.

Reply:
Not sure what info we need to grab from the route (hostname?)